contracts/optimistic-ethereum/OVM/verification/OVM_StateTransitioner.sol

```diff
@@ -266,32 +263,38 @@ contract OVM_StateTransitioner is Lib_AddressResolver, OVM_FraudContributor, iOV
             "Contract must be verified before proving a storage slot."
         );
 
-        (
-            bool exists,
-            bytes memory encodedValue
-        ) = Lib_SecureMerkleTrie.get(
-            abi.encodePacked(_key),
-            _storageTrieWitness,
-            ovmStateManager.getAccountStorageRoot(_ovmContractAddress)
-        );
+        bytes32 storageRoot = ovmStateManager.getAccountStorageRoot(_ovmContractAddress);
+        bytes32 value;
 
-        if (exists == true) {
-            bytes memory value = Lib_RLPReader.readBytes(encodedValue);
-            require(
-                keccak256(value) == keccak256(abi.encodePacked(_value)),
-                "Provided storage slot value is invalid."
-            );
+        if (storageRoot == EMPTY_ACCOUNT_STORAGE_ROOT) {
+            // Storage trie was empty, so the user is always allowed to insert zero-byte values.
+            value = bytes32(0);
         } else {
-            require(
-                _value == bytes32(0),
-                "Provided storage slot value is invalid."
+            (
+                bool exists,
+                bytes memory encodedValue
+            ) = Lib_SecureMerkleTrie.get(
```

Comment:
recommend we duplicate the "this call will revert" explanation similar to the comment above.

Reply:
Adding.